server/src/main/java/org/elasticsearch/action/fieldcaps/TransportFieldCapabilitiesAction.java

```diff
@@ -167,6 +167,14 @@ private void doExecuteForked(Task task, FieldCapabilitiesRequest request, final
                     resp = new FieldCapabilitiesIndexResponse(resp.getIndexName(), curr.getIndexMappingHash(), curr.get(), true);
                 }
             }
+            indexResponses.merge(resp.getIndexName(), resp, (a, b) -> {
```

Comment:
You only need to run this merge `request.includeFieldsWithNoValue() == false` and the line below `indexResponses.putIfAbsent(resp.getIndexName(), resp);` should be run if it's true right?

Reply:
That's correct. Changing it :)